presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -912,27 +1087,68 @@ public int hashCode()
         return Objects.hash(id);
     }
 
+    public InternalResourceGroup getRoot()
+    {
+        synchronized (root) {
```

Comment:
`root` would not change and I can't think why its access needs to be serialized with other methods so why does this need to be synchronized?

Reply:
yes not needed, remove it.